delta/kernel/src/main/scala/ch/epfl/bluebrain/nexus/delta/kernel/effect/migration/MigrateEffectSyntax.scala

```diff
@@ -34,7 +36,19 @@ final class MonixBioToCatsIOEitherOps[E, A](private val io: BIO[E, A]) extends A
 }
 
 final class CatsIOToBioOps[A](private val io: IO[A]) extends AnyVal {
-  def toBIO[E <: Throwable](implicit E: ClassTag[E]): BIO[E, A] =
+
+  /**
+    * Safe conversion between CE and Monix, forcing the user to specify a strict subtype of [[Throwable]]. If omitted,
+    * the compiler may infer [[Throwable]] and bypass any custom error handling.
+    */
+  @nowarn
+  def toBIO[E <: Throwable](implicit E: ClassTag[E], ev: E =:!= Throwable): BIO[E, A] =
```

Comment:
Think we should prefer this generally, particularly when migrating all of the SDK since there's a lot of custom error handling. It's a bit scary not having it in the type system otherwise 😅